frontend/src/components/Recipes/Add/RecipeForm.tsx

```diff
@@ -1,42 +1,184 @@
+    },
+    {
+        id: 'ingredients',
+        name: 'ingredients',
+        type: 'text',
+        label: 'Ingredients',
+        placeholder: 'Insert ingredients',
+        required: true,
+        pattern: '^[a-zA-Z0-9\s]+$',
+        errorMessage:
+            'Ingredients must contain only letters, numbers and spaces',
+    },
+    {
+        id: 'instructions',
+        name: 'instructions',
+        type: 'text',
+        label: 'Instructions',
+        placeholder: 'Insert instructions',
+        required: true,
+        pattern: '^[a-zA-Z0-9\s]+$',
+        errorMessage:
+            'Instructions must contain only letters, numbers and spaces',
+    },
+    {
+        id: 'cuisineId',
+        name: 'cuisineId',
+        type: 'number',
+        label: 'Cuisine',
+        placeholder: 'Insert cuisine',
+        required: true,
+        min: 1,
+        max: 3,
+        errorMessage: 'Cuisine must be a number between 1 and 3',
+    },
+    {
+        id: 'dietId',
+        name: 'dietId',
+        type: 'number',
+        label: 'Dietary',
+        placeholder: 'Insert dietary',
+        required: true,
+        min: 1,
+        max: 3,
+        errorMessage: 'Dietary must be a number between 1 and 3',
+    },
+    {
+        id: 'difficultyId',
+        name: 'difficultyId',
+        type: 'number',
+        label: 'Difficulty',
+        placeholder: 'Insert difficulty',
+        required: true,
+        min: 1,
+        max: 3,
+        errorMessage: 'Difficulty must be a number between 1 and 3',
+    },
+    {
+        id: 'image',
+        name: 'image',
+        type: 'file',
+        label: 'Image',
+        placeholder: 'Insert image',
+        required: true,
+        accept: 'image/png, image/jpeg, image/jpg',
+        errorMessage: 'Image must be a png, jpeg or jpg',
+    },
+]
+
+const initialValues = {
+    name: '',
+    ingredients: '',
+    instructions: '',
+    cuisineId: 1,
+    dietId: 1,
+    difficultyId: 1,
+    image: '',
+}
+
+const initialErrorVisibility = {
+    name: false,
+    ingredients: false,
+    instructions: false,
+    cuisineId: false,
+    dietId: false,
+    difficultyId: false,
+    image: false,
+}
+
 export default function RecipeForm() {
+    const [values, setValues] = useState<FormInputTypes>(initialValues)
+    const [errorVisibility, setErrorVisibility] = useState(
+        initialErrorVisibility
+    )
+
+    function handleChange(e: React.ChangeEvent<HTMLInputElement>) {
+        setValues({ ...values, [e.target.name]: e.target.value })
+    }
+
+    async function handleSubmit(e: React.FormEvent<HTMLFormElement>) {
+        e.preventDefault()
+
+        const form = new FormData(e.currentTarget)
+
+        const sendData = await fetch(`${endpoint}/recipes`, {
+            method: 'POST',
+            headers: {
+                enctype: 'multipart/form-data',
+            },
+            body: form,
+        })
```

Comment:
I would define this method outside of the handle submit, even in another file, where you define the function that performs the POST request